### ISSUE TITLE
Adapt to SD-v3's VAE

### ISSUE THREAD

**Issue by k-zha14** [PixArt-alpha/PixArt-sigma#114]
Hi, great work! I succeed in reproducing the VAE adaption from SD2' to SDXL's, as discussed in Pixart-Sigma. However, the adaption to SDXL's VAE is not successful. After 10k steps finetuning in SAM,  sampled images are meaningless and chaotic(attached below), although the training loss looks pretty good. 

![image (3)](https://github.com/PixArt-alpha/PixArt-sigma/assets/32434681/f2925425-c8c3-4e19-9f89-40fe5d86a4d3)
![image (2)](https://github.com/PixArt-alpha/PixArt-sigma/assets/32434681/100feb95-930c-48cb-888a-ab761219b235)
![image (1)](https://github.com/PixArt-alpha/PixArt-sigma/assets/32434681/5cd7108a-e201-4289-b93e-9c5c5495bdb4)
![image](https://github.com/PixArt-alpha/PixArt-sigma/assets/32434681/d6bc7988-0a90-4496-bb3d-3eae40a25e50)
(the first two images generated from the the adaption exp. - SD2's to SDXL's VAE, while the latter from SD2' to SDv3's VAE)

The key change of SDv3's VAE is, the latent channel expands from 4 to 16. Thus, the compressed latents will reserve more details and avoid unplesant artifacts(eg. little faces, texts). To accommodate with this change, I initialize the net with official 'Pixart-alpha-256x256.pt' weights, except the 'x_embed' layer and 'final_layer'(channels 4->16, 8 -> 32).

Could anyone give me some hints? I'm really confused. Thanks, guys!

**Comment by ReyJ94** [comment]
i think the devs are working on a new similar vae. Will see

**Comment by lawrence-cj** [comment]
Have you changed the vae's scale_factor accordingly?

**Comment by k-zha14** [comment]
> Have you changed the vae's scale_factor accordingly?

Absolutely! Specifically, the SDv3's VAE adds 'scale_factor'. So the normalization of VAE's latent should be:
'''
latents = (latents - self.config.shift_factor) * self.config.scaling_factor

Waiting for more progress! Let's work together to make the community more prosperous. CC @ReyJ94 @lawrence-cj

**Comment by Feynman1999** [comment]
> > Have you changed the vae's scale_factor accordingly?
> 
> Absolutely! Specifically, the SDv3's VAE adds 'scale_factor'. So the normalization of VAE's latent should be:
> 
> `latents = (latents - self.config.shift_factor) * self.config.scaling_factor`
> 
> Waiting for more progress! Let's work together to make the PixArt community more prosperous. CC @ReyJ94 @lawrence-cj

Have you successfully adapted to SD3 VAE? I have been trying recently, but have not been successful
